src/PaymentProcessing/SpherePayment.php

```diff
@@ -0,0 +1,257 @@
+        if ($this->front == 'patient') {
+            $this->trxcustid = $cryptoGen->decryptStandard($GLOBALS['sphere_patientfront_trxcustid']);
+            $this->trxcustidLicensekey = $cryptoGen->decryptStandard($GLOBALS['sphere_patientfront_trxcustid_licensekey']);
+            if ($this->testing) {
+                $url = self::PATIENTFRONT_TESTING_URL;
+            } else {
+                $url = self::PATIENTFRONT_PRODUCTION_URL;
+            }
+        } else { //$this->front == 'clinic'
+            $this->trxcustid = $cryptoGen->decryptStandard($GLOBALS['sphere_clinicfront_trxcustid']);
+            $this->trxcustidLicensekey = $cryptoGen->decryptStandard($GLOBALS['sphere_clinicfront_trxcustid_licensekey']);
+            if ($this->testing) {
+                $url = self::CLINICFRONT_TESTING_URL;
+            } else {
+                $url = self::CLINICFRONT_PRODUCTION_URL;
+            }
+        }
+
+        // Calculate the OpenEMR server
+        $this->serverSite = $GLOBALS['site_addr_oath'] . $GLOBALS['web_root'];
+
+        // Calculate the $mainUrl
+        $this->mainUrl = $url . '?trxcustid=' . urlencode($this->trxcustid) . '&trxcustid_licensekey=' . urlencode($this->trxcustidLicensekey) . '&trxcustomfield[1]=' . urlencode($this->front) . '&trxcustomfield[2]=' . urlencode($this->patientIdCc) . '&trxcustomfield[3]=' . urlencode(CsrfUtils::collectCsrfToken('sphere'));
+
+        if ($this->front == 'patient') {
+            // Add specific parameters for patient front to the $mainUrl
+            $this->mainUrl = $this->mainUrl . '&hide_ticket=y&cvc_help=y&avs=y&postal=y&fulladdress=y&show_email=y';
+        } else {
+            // Add specific parameters for clinic front to the $mainUrl
+            $operatorEntry = (new UserService())->getCurrentlyLoggedInUser();
+            $operatorName = $operatorEntry['fname'] . " " . $operatorEntry['lname'];
+            $this->mainUrl = $this->mainUrl . '&hide_ticket=y&hide_trxoperator=y&trxoperator=' . urlencode($operatorName);
+        }
+    }
+
+    /**
+     * Return sphere code in javascript block.
+     *
+     * @return string
+     */
+    public function renderSphereJs(): string
+    {
+        if ($this->front == 'patient') {
+            return '<script>' . $this->renderSphereJsPatientFront() . $this->renderSphereJsCore() . '</script>';
+        } else { // $this->front == 'clinic'
+            return '<script>' . $this->renderSphereJsClinicFront() . $this->renderSphereJsCore() . '</script>';
+        }
+    }
+
+    /**
+     * Return patient front specific sphere code in javascript block.
+     *
+     * @return string
+     */
+    private function renderSphereJsPatientFront(): string
+    {
+        return "
+            function sphereSuccess(encData) {
+                let oForm = document.forms['payment-form'];
+                oForm.elements['mode'].value = 'Sphere';
+
+                let inv_values = JSON.stringify(getFormObj('invoiceForm'));
```

Comment:
Do you care if someone inserts a javascript value that can't be stringified here?  I usually put these in try catch blocks.